packages/ra-core/src/types.ts

```diff
@@ -68,8 +68,8 @@ export interface UserIdentity {
 export type AuthProvider = {
     login: (params: any) => Promise<any>;
     logout: (params: any) => Promise<void | string>;
-    checkAuth: (params: any) => Promise<void>;
-    checkError: (error: any) => Promise<void>;
+    checkAuth: (params: any) => Promise<any>;
+    checkError: (error: any) => Promise<any>;
```

Comment:
Then a generic type with a default to `void` might be better don't you think?

Reply:
Somehing like:
```ts
checkError: <T = void>(error: any) => Promise<T>;
```

Reply:
@djhi I tried the following initially. However, I still get a type error when adding the authProvider
```typescript
checkError: <T = any>(error: any) => Promise<T>;
```

I get the following linting error the same applies for `void`
```
Type 'Promise<User>' is not assignable to type 'Promise<T>'.
        Type 'User' is not assignable to type 'T'.
          'T' could be instantiated with an arbitrary type which could be unrelated to 'User'
```

I could get around this by doing the following.
```tsx
 <Admin
        dataProvider={dataProvider}
        authProvider={authProvider as AuthProvider}
      >
```

It's your call, but just some AuthProviders might need to have to be overriden.

Let me know what you think.

Reply:
At least the override would work. With `Promise<void>` the override errors out saying the types are incompatible.